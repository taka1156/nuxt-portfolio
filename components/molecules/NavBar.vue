<template>
  <div>
    <div class="nav-bar">
      <div class="nav-bar__box">
        <nav-logo>{{ logoText }}</nav-logo>
        <div @click="$emit('change-state')">
          <nav-icon :is-open="isOpen" />
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import NavIcon from '../atoms/NavIcon';
import NavLogo from '../atoms/NavLogo';

export default {
  name: 'NavBar',
  components: {
    'nav-icon': NavIcon,
    'nav-logo': NavLogo,
  },
  props: {
    logoText: {
      type: String,
      default: 'Please Setting Logo',
      required: true,
    },
    isOpen: {
      type: Boolean,
      default: false,
      required: true,
    },
  },
};
</script>

<style scoped>
.nav-bar {
  position: fixed;
  top: 0;
  left: 0;
  height: 60px;
  width: 100%;
  background-color: rgba(12, 12, 12, 0.9);
  z-index: 2;
}

.nav-bar__box {
  display: flex;
  justify-content: space-between;
  height: 90%;
  width: 100%;
  margin: 0.5em auto;
}
</style>
