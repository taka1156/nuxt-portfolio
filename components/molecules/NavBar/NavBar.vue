<template>
  <div>
    <div class="nav-bar">
      <div class="nav-bar__box">
        <base-link :route-to="`/`">{{ logoText }}</base-link>
        <base-nav-icon :is-open="isOpen" @btn-click="changeState">
          {{ isOpen ? 'CLOSE' : 'NAVI' }}
        </base-nav-icon>
      </div>
    </div>
  </div>
</template>

<script>
import BaseLink from '../../atoms/BaseLink/BaseLink';
import BaseNavIcon from '../../atoms/BaseNavIcon/BaseNavIcon';

export default {
  name: 'NavBar',
  components: {
    'base-nav-icon': BaseNavIcon,
    'base-link': BaseLink
  },
  props: {
    /**
     * ロゴテキスト
     */
    logoText: {
      type: String,
      required: true
    },
    /**
     * ナビゲーションの開閉状態
     */
    isOpen: {
      type: Boolean,
      required: true
    }
  },
  methods: {
    changeState() {
      /**
       * ナビゲーション開閉の状態を変化させる
       * (開く)
       * @event changeState
       */
      this.$emit('change-state', !this.isOpen);
    }
  }
};
</script>

<style scoped>
::v-deep .base-link--extend {
  display: block;
  height: 30px;
  margin-top: 10px;
  font-size: 1.3em;
  color: white;
}

.nav-bar {
  position: fixed;
  top: 0;
  left: 0;
  z-index: 2;
  width: 100%;
  height: 60px;
  background-color: rgba(12, 12, 12, 0.9);
}

.nav-bar__box {
  display: flex;
  justify-content: space-between;
  width: 100%;
  height: 85%;
  margin: 0.5em auto;
}
</style>
