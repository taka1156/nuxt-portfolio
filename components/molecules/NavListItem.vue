<template>
  <div>
    <li class="nav__item" @click="$emit('change-state')">
      <nuxt-link :to="navItem.to" class="nav__link">
        <nav-img :nav-img="navItem.img" />
        <nav-text :nav-text="navItem.name" />
      </nuxt-link>
    </li>
  </div>
</template>

<script>
import NavImg from '../atoms/NavImg';
import NavText from '../atoms/NavText';

export default {
  name: 'NavListItem',
  components: {
    'nav-img': NavImg,
    'nav-text': NavText,
  },
  props: {
    navItem: {
      type: Object,
      default: () => {},
      required: true,
    },
  },
};
</script>

<style scoped>
.nav__item {
  display: block;
  font-size: 50px;
  text-align: left;
  color: white;
}

.nav__link {
  display: flex;
  width: 100%;
  text-decoration: none;
}
</style>
