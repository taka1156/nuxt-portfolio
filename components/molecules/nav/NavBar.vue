<template>
  <div>
    <div class="nav-bar">
      <div class="nav-bar__box">
        <nav-logo>{{ logoText }}</nav-logo>
        <base-btn :use-type="`none`" @btn-click="changeState">
          <nav-icon :is-open="isOpen" />
        </base-btn>
      </div>
    </div>
  </div>
</template>

<script>
import NavLogo from '../../atoms/nav/NavLogo';
import NavIcon from '../../atoms/nav/NavIcon';
import BaseBtn from '../../atoms/base/BaseBtn.vue';
export default {
  name: 'NavBar',
  components: {
    'nav-icon': NavIcon,
    'nav-logo': NavLogo,
    'base-btn': BaseBtn,
  },
  props: {
    /**
     * ロゴテキスト
     */
    logoText: {
      type: String,
      default: 'Please Setting Logo',
      required: true,
    },
    /**
     * ナビゲーションの開閉状態
     */
    isOpen: {
      type: Boolean,
      default: false,
      required: true,
    },
  },
  methods: {
    changeState() {
      /**
       * ナビゲーション開閉の状態を変化させる
       * (開く)
       * @event changeState
       */
      this.$emit('change-state', !this.isOpen);
    },
  },
};
</script>

<style scoped>
.nav-bar {
  position: fixed;
  top: 0;
  left: 0;
  height: 60px;
  width: 100%;
  background-color: rgba(40, 167, 69, 0.9);
  z-index: 2;
}
.nav-bar__box {
  display: flex;
  justify-content: space-between;
  height: 90%;
  width: 100%;
  margin: 0.5em auto;
}
</style>
