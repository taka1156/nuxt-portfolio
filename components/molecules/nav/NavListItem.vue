<template>
  <div>
    <li class="nav-list-item" @click="$emit('change-state')">
      <nuxt-link :to="navItem.to" class="nav-list-item__link">
        <nav-img :nav-img="navItem.img" :img-alt="`${navItem.name}の画像`" />
        <base-text class="base-text--nav">{{ navItem.name }}</base-text>
      </nuxt-link>
    </li>
  </div>
</template>

<script>
import NavImg from '../../atoms/nav/NavImg';
import BaseText from '../../atoms/base/BaseText';

export default {
  name: 'NavListItem',
  components: {
    'nav-img': NavImg,
    'base-text': BaseText,
  },
  props: {
    navItem: {
      type: Object,
      default: () => {},
      required: true,
    },
  },
};
</script>

<style scoped>
.nav-list-item {
  display: block;
  font-size: 50px;
  text-align: left;
  color: white;
}

.nav-list-item__link {
  display: flex;
  width: 100%;
  text-decoration: none;
}
</style>
