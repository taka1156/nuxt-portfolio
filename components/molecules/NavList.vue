<template>
  <div>
    <transition name="nav-fade">
      <div v-show="isOpen" class="nav__list">
        <ul>
          <nav-item
            v-for="(item, index) in routers"
            :key="index"
            class="nav__item"
            :nav-item="item"
            @change-state="$emit('change-state')"
          />
        </ul>
      </div>
    </transition>
  </div>
</template>

<script>
import NavListItem from './NavListItem';

export default {
  name: 'NavList',
  components: {
    'nav-item': NavListItem,
  },
  props: {
    isOpen: {
      type: Boolean,
      default: false,
      required: true,
    },
    routers: {
      type: Array,
      default: () => [],
      required: true,
    },
  },
};
</script>

<style scoped>
/* css reset */
ul {
  margin: 0px;
  padding: 0px;
}
/* css reset */

.nav__list {
  display: flex;
  justify-content: center;
  position: fixed;
  top: 60px;
  right: 0;
  left: 0;
  height: 100vh;
  width: 100vw;
  z-index: 2;
  background-color: rgba(12, 12, 12, 0.9);
}

/* nav-contentsのアニメーション */
.nav-fade-enter,
.nav-fade-leave-to {
  opacity: 0;
}

.nav-fade-enter-to,
.nav-fade-leave {
  opacity: 1;
}

.nav-fade-enter-active,
.nav-fade-leave-active {
  transition: opacity 0.6s ease;
}
</style>
