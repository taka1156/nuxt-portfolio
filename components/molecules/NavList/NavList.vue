<template>
  <div>
    <transition name="nav-fade">
      <div v-show="isOpen" class="nav-list">
        <ul>
          <li
            v-for="(item, index) in routes"
            :key="`nav_${index}`"
            @click="changeState"
          >
            <nav-list-item :nav-item="item" @change-state="changeState" />
          </li>
        </ul>
      </div>
    </transition>
  </div>
</template>

<script>
import NavListItem from '../NavListItem/NavListItem';

export default {
  name: 'NavList',
  components: {
    'nav-list-item': NavListItem
  },
  props: {
    /**
     * ナビゲーションの開閉状態
     */
    isOpen: {
      type: Boolean,
      required: true
    },
    /**
     * ナビゲーションの項目が定義された配列
     */
    routes: {
      type: Array,
      required: true
    }
  },
  methods: {
    changeState() {
      /**
       * ナビゲーション開閉の状態を変化させる
       * (閉じる)
       * @event changeState
       */
      this.$emit('change-state', false);
    }
  }
};
</script>

<style scoped>
/* css reset */
ul,
li {
  list-style-type: none;
}

/* css reset */
.nav-list {
  position: fixed;
  top: 60px;
  right: 0;
  left: 0;
  z-index: 3;
  display: flex;
  justify-content: center;
  width: 100vw;
  height: 100vh;
  background-color: rgba(12, 12, 12, 0.9);
}

/* nav-contentsのアニメーション */
.nav-fade-enter,
.nav-fade-leave-to {
  opacity: 0;
}

.nav-fade-enter-to,
.nav-fade-leave {
  opacity: 1;
}

.nav-fade-enter-active,
.nav-fade-leave-active {
  transition: opacity 0.6s ease;
}
</style>
