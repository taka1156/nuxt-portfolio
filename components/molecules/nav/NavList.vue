<template>
  <div>
    <transition name="nav-fade">
      <div v-show="isOpen" class="nav-list">
        <ul>
          <nav-item
            v-for="(item, index) in routes"
            :key="index"
            :nav-item="item"
            @change-state="changeState"
          />
        </ul>
      </div>
    </transition>
  </div>
</template>

<script>
import NavListItem from './NavListItem';
export default {
  name: 'NavList',
  components: {
    'nav-item': NavListItem,
  },
  props: {
    /**
     * ナビゲーションの開閉状態
     */
    isOpen: {
      type: Boolean,
      default: false,
      required: true,
    },
    /**
     * ナビゲーションの項目が定義された配列
     */
    routes: {
      type: Array,
      default: () => [],
      required: true,
    },
  },
  methods: {
    changeState() {
      /**
       * ナビゲーション開閉の状態を変化させる
       * (閉じる)
       * @event changeState
       */
      this.$emit('change-state', false);
    },
  },
};
</script>

<style scoped>
.nav-list {
  display: flex;
  justify-content: center;
  position: fixed;
  top: 60px;
  right: 0;
  left: 0;
  height: 100vh;
  width: 100vw;
  z-index: 2;
  background-color: rgba(40, 167, 69, 0.9);
}

/* nav-contentsのアニメーション */
.nav-fade-enter,
.nav-fade-leave-to {
  opacity: 0;
}
.nav-fade-enter-to,
.nav-fade-leave {
  opacity: 1;
}
.nav-fade-enter-active,
.nav-fade-leave-active {
  transition: opacity 0.6s ease;
}
</style>
