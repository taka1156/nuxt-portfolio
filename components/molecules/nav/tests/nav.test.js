import { mount, RouterLinkStub } from '@vue/test-utils';
import NavBar from '../NavBar.vue';
import NavList from '../NavList.vue';
import NavListItem from '../NavListItem.vue';
import { dummyLogo, dummyRoutes } from '@/__testdata__/testdata.js';

// NavBar
describe('NavBar', () => {
  const navBar = propsData => {
    return mount(NavBar, {
      propsData: {
        ...propsData,
        logoText: dummyLogo,
      },
    });
  };

  it('NavBar初期値: isOpen, logoText', () => {
    const wrapper = navBar({ isOpen: false });
    // logoText
    expect(wrapper.vm.$options.props.logoText.required).toBe(true);
    expect(wrapper.vm.logoText).toBe(dummyLogo);
    // isOpen
    expect(wrapper.vm.$options.props.isOpen.required).toBe(true);
    expect(wrapper.vm.isOpen).toBe(false);
  });

  it('logoTextが反映されているか', () => {
    const wrapper = navBar({ isOpen: false });
    const navLogo = wrapper.find('.nav-logo');
    expect(navLogo.text()).toBe(dummyLogo);
  });

  it('ナビゲーションバーを開く', () => {
    const wrapper = navBar({ isOpen: true });
    expect(wrapper.vm.isOpen).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('ナビゲーションバーを閉じる', () => {
    const wrapper = navBar({ isOpen: false });
    expect(wrapper.vm.isOpen).toBe(false);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('changeStateを実行した時、`change-state`のイベントが発生するか', () => {
    const wrapper = navBar({ isOpen: false });
    wrapper.vm.changeState();
    expect(wrapper.emitted('change-state')).toBeTruthy();
  });
});

// NavList
describe('NavList', () => {
  const navList = propsData => {
    return mount(NavList, {
      stubs: {
        NuxtLink: RouterLinkStub,
      },
      propsData: {
        routes: dummyRoutes,
        ...propsData,
      },
    });
  };

  it('NavList初期値: isOpen, routes', () => {
    const wrapper = navList({ isOpen: false });
    // isOpen
    expect(wrapper.vm.$options.props.isOpen.required).toBe(true);
    expect(wrapper.vm.isOpen).toBe(false);
    // routes
    expect(wrapper.vm.$options.props.routes.required).toBe(true);
    console.log(wrapper.vm.$options.props.routes.default());
    expect(wrapper.vm.routes).toBe(dummyRoutes);
  });

  it('値がDOMに反映されているか', () => {
    const wrapper = navList({ isOpen: false });
    const aTag = wrapper.findComponent(RouterLinkStub);
    const imgTag = wrapper.find('img');

    expect(aTag.props().to).toBe(dummyRoutes[0].to);
    expect(aTag.text()).toBe(dummyRoutes[0].name);
    expect(imgTag.attributes().src).toBe(dummyRoutes[0].img);
  });

  it('isOpenがtrueの時に`.nav-list`が表示', () => {
    const wrapper = navList({ isOpen: true });
    expect(wrapper.find('.nav-list').element.style.display).not.toBe('none');
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('isOpenがfalseの時に`.nav-list`が非表示', () => {
    const wrapper = navList({ isOpen: false });
    expect(wrapper.find('.nav-list').element.style.display).toBe('none');
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('changeStateを実行した時、`change-state`のイベントが発生するか', () => {
    const wrapper = navList({ isOpen: true });
    wrapper.vm.changeState();
    expect(wrapper.emitted('change-state')).toBeTruthy();
  });
});

// NavListItem
describe('NavListItem', () => {
  const navListItem = mount(NavListItem, {
    // <nuxt-link>をRouterLinkStubで探せるようになる
    stubs: {
      NuxtLink: RouterLinkStub,
    },
    propsData: {
      navItem: dummyRoutes[0],
    },
  });

  it('NavList初期値: navItem', () => {
    // NavItme
    expect(navListItem.vm.$options.props.navItem.required).toBe(true);
    console.log(navListItem.vm.$options.props.navItem.default());
    expect(navListItem.vm.navItem).toBe(dummyRoutes[0]);
  });

  it('値がDOMに反映されているか', () => {
    const aTag = navListItem.findComponent(RouterLinkStub);
    const imgTag = navListItem.find('img');

    expect(aTag.props().to).toBe(dummyRoutes[0].to);
    expect(aTag.text()).toBe(dummyRoutes[0].name);
    expect(imgTag.attributes().src).toBe(dummyRoutes[0].img);
    // スナップショット
    expect(navListItem.html()).toMatchSnapshot();
  });

  it('changeStateを実行した時、`change-state`のイベントが発生するか', () => {
    navListItem.vm.changeState();
    expect(navListItem.emitted('change-state')).toBeTruthy();
  });
});
