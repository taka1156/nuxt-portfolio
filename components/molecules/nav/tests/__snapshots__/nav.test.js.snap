// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar ナビゲーションバーを閉じる 1`] = `
"<div>
  <div class=\\"nav-bar\\">
    <div class=\\"nav-bar__box\\">
      <div><a href=\\"/\\" class=\\"nav-logo\\">ダミーロゴ</a></div>
      <div><button class=\\"base-btn base-btn--none\\">
          <div>
            <div class=\\"nav-icon\\"><span class=\\"nav-icon__top nav-icon__top--close\\"></span> <span class=\\"nav-icon__middle\\"></span> <span class=\\"nav-icon__bottom nav-icon__bottom--close\\"></span></div>
          </div>
        </button></div>
    </div>
  </div>
</div>"
`;

exports[`NavBar ナビゲーションバーを開く 1`] = `
"<div>
  <div class=\\"nav-bar\\">
    <div class=\\"nav-bar__box\\">
      <div><a href=\\"/\\" class=\\"nav-logo\\">ダミーロゴ</a></div>
      <div><button class=\\"base-btn base-btn--none\\">
          <div>
            <div class=\\"nav-icon\\"><span class=\\"nav-icon__top nav-icon__top--open\\"></span> <span class=\\"nav-icon__middle nav-icon__middle--fade\\"></span> <span class=\\"nav-icon__bottom nav-icon__bottom--open\\"></span></div>
          </div>
        </button></div>
    </div>
  </div>
</div>"
`;

exports[`NavList isOpenがfalseの時に\`.nav-list\`が非表示 1`] = `
"<div>
  <transition-stub name=\\"nav-fade\\">
    <div class=\\"nav-list\\" style=\\"display: none;\\">
      <ul>
        <div>
          <li class=\\"nav-item\\"><a class=\\"nav-item__link\\">
              <div><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミーナビゲーションの画像\\" class=\\"base-img base-img--lg\\"></div>
              <p class=\\"base-text base-text--nav\\">ダミーナビゲーション</p>
            </a></li>
        </div>
        <div>
          <li class=\\"nav-item\\"><a class=\\"nav-item__link\\">
              <div><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミーナビゲーションの画像\\" class=\\"base-img base-img--lg\\"></div>
              <p class=\\"base-text base-text--nav\\">ダミーナビゲーション</p>
            </a></li>
        </div>
        <div>
          <li class=\\"nav-item\\"><a class=\\"nav-item__link\\">
              <div><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミーナビゲーションの画像\\" class=\\"base-img base-img--lg\\"></div>
              <p class=\\"base-text base-text--nav\\">ダミーナビゲーション</p>
            </a></li>
        </div>
        <div>
          <li class=\\"nav-item\\"><a class=\\"nav-item__link\\">
              <div><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミーナビゲーションの画像\\" class=\\"base-img base-img--lg\\"></div>
              <p class=\\"base-text base-text--nav\\">ダミーナビゲーション</p>
            </a></li>
        </div>
      </ul>
    </div>
  </transition-stub>
</div>"
`;

exports[`NavList isOpenがtrueの時に\`.nav-list\`が表示 1`] = `
"<div>
  <transition-stub name=\\"nav-fade\\">
    <div class=\\"nav-list\\">
      <ul>
        <div>
          <li class=\\"nav-item\\"><a class=\\"nav-item__link\\">
              <div><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミーナビゲーションの画像\\" class=\\"base-img base-img--lg\\"></div>
              <p class=\\"base-text base-text--nav\\">ダミーナビゲーション</p>
            </a></li>
        </div>
        <div>
          <li class=\\"nav-item\\"><a class=\\"nav-item__link\\">
              <div><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミーナビゲーションの画像\\" class=\\"base-img base-img--lg\\"></div>
              <p class=\\"base-text base-text--nav\\">ダミーナビゲーション</p>
            </a></li>
        </div>
        <div>
          <li class=\\"nav-item\\"><a class=\\"nav-item__link\\">
              <div><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミーナビゲーションの画像\\" class=\\"base-img base-img--lg\\"></div>
              <p class=\\"base-text base-text--nav\\">ダミーナビゲーション</p>
            </a></li>
        </div>
        <div>
          <li class=\\"nav-item\\"><a class=\\"nav-item__link\\">
              <div><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミーナビゲーションの画像\\" class=\\"base-img base-img--lg\\"></div>
              <p class=\\"base-text base-text--nav\\">ダミーナビゲーション</p>
            </a></li>
        </div>
      </ul>
    </div>
  </transition-stub>
</div>"
`;

exports[`NavListItem 値がDOMに反映されているか 1`] = `
"<div>
  <li class=\\"nav-item\\"><a class=\\"nav-item__link\\">
      <div><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミーナビゲーションの画像\\" class=\\"base-img base-img--lg\\"></div>
      <p class=\\"base-text base-text--nav\\">ダミーナビゲーション</p>
    </a></li>
</div>"
`;
